functions/functionsv2/helloauditlog/hello_auditlog.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_log_cloudevent]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+// AuditLogEntry represents a LogEntry as described at
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+type AuditLogEntry struct {
+	ProtoPayload *AuditLogProtoPayload `json:"protoPayload"`
+}
+
+// AuditLogProtoPayload represents AuditLog within the LogEntry.protoPayload
+// See https://cloud.google.com/logging/docs/reference/audit/auditlog/rest/Shared.Types/AuditLog
+type AuditLogProtoPayload struct {
+	MethodName         string                 `json:"methodName"`
+	ResourceName       string                 `json:"resourceName"`
+	AuthenticationInfo map[string]interface{} `json:"authenticationInfo"`
+}
+
+// HelloAuditLog receives a CloudEvent containing an AuditLogEntry, and logs a few fields.
+func HelloAuditLog(ctx context.Context, e event.Event) error {
+	// Print out details from the CloudEvent itself
+	// See https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#subject
+	// for details on the Subject field
+	log.Printf("Event Type: %s", e.Type())
+	log.Printf("Subject: %s", e.Subject())
+
+	// Decode the Cloud Audit Logging message embedded in the CloudEvent
+	logentry := &AuditLogEntry{}
+	if err := e.DataAs(logentry); err != nil {
+		ferr := fmt.Errorf("event.DataAs: %v", err)
```

Comment:
nit: Maybe this should wrap the error with %w?

Reply:
Done!